tests/neg/i974.scala

```diff
@@ -0,0 +1,8 @@
+trait Foo[T <: Bar[T]#Elem] // error: illegal cyclic reference
+trait Bar[T] {
+  type Elem = T
+}
+trait Foo2[T <: Bar2[T]#Elem] // error: illegal cyclic reference
```

Comment:
Did you intend to have the same test twice, with just the names changed? (`Foo -> Foo2`, `Bar -> Bar2`)